### DISCUSSION TITLE
Harman splash screen using ADL

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#2073] [Q&A] [answered]
Hello,

on Windows, when I use ADL the Harman screen appear when the app start, it's not appear with a captive runtime, I have a license it's strange ?

Thanks

**Comment by ajwfrost** [comment, marked as answer]
If you have a license, and the splash screen is appearing just when using ADL, then it's most likely because of a bug in how that's looking for the license file. We moved the file location, ADL should be able to find the new file but it still looks first in the old location, so there may be an out-dated file left there..

So can you please check:
- in the SDK location "lib" folder, there's an adt.lic file: what's the date-stamp at the top of the file?
- and in your home folder (/Users/username or c:\users\username) there should be a folder ".airsdk" with adt.lic in there: can you please also check the date-stamp at the top of the file?

Just to check it is what I think it is: if the "lib" location has an older file, then try copying the file from the home folder into that location and try it again. If that solved it, then you are best off deleting the file from the "lib" folder since ADL should then start picking up the file directly from the ".airsdk" folder.

thanks

**Comment by pol2095** [comment]
You're right, the file in "lib" the date-stamp is 2020 and in ".airsdk" the date-stamp is 2022, thanks.

**Comment by 2jfw** [comment]
We are having the same issue where we cannot get rid of the splash though the licenses of our team are valid - we cannot seem to sort it out.

Scenario is:
- `.airsdk` (home) folder was empty; I copied `adt.lic` file from `airsdk/lib`  folder to that location
- `adt.bat -license` states: **AIR Developer Tool version 33.1.1.795**
- Java HOME in `adt.cfg` is set to `JAVA_HOME=C:\\Program Files\\Java\\jre1.8.0_333\\bin` (folders and files exist)
- Java HOME is correctly set in environmental variables (`JAVA_HOME` and also in `Path`)

What are we missing?

**Reply by ajwfrost** [reply]
Does the `adt.bat -license` call tell you that the license is valid? and does it say where the file was that it checked?

Is the splash screen happening just when you're debugging (ADL), or is it appearing in applications that you package up with ADT as well?

And if you have an adt.lic file in both the `airsdk/lib` folder, and the `~/.airsdk` folder, can you please check what the datestamps are that are the top of each file? (and you can probably then delete the older one..)

thanks

**Reply by 2jfw** [reply]
Oh, wait - the checking of the license file with adt.bat did actually took a while after tracing "AIR Developer Tool version 33.1.1.795" (so no instant output) - that's why I missed the following obvious error:

```
WARNING: ADT license file for developer ID "_XXX_" reported as: ERROR
License file has been removed: please contact adobe.support@harman.com to renew your license
WARNING - ADT license has expired, splash screen will be injected
License file was present but not valid
DeveloperID=_XXX_
```

I replaced the developer ID with _XXX_

**Reply by 2jfw** [reply]
Just tested `adt-bat -license` again and the time between output `AIR Developer Tool version 33.1.1.795` and the actual error `WARNING: ADT license file for developer  [...]` is around **21s** (so thats why I missed it earlier..)

**Reply by 2jfw** [reply]
Our licenses should actually still be valid (date issued is `#Tue Nov 30 10:06:35 UTC 2021`)

**Reply by ajwfrost** [reply]
That sounds like a network issue then, that's a lot longer timeout than I've come across before! usually I thought it was max 3s.. ("ERROR" = something went wrong whilst checking; "INVALID" would be shown if we checked and the developer ID wasn't actually valid...)

Do you have any network restrictions on https access? it's trying to get to an AWS server to check on the license status..

thanks

**Reply by 2jfw** [reply]
Okay, thanks for your input, Andrew! I will talk to the IT and see if they have changed network related settings (/security), since the license check was actually working fine until a few weeks ago, where it started showing the splash ..

**Reply by 2jfw** [reply]
So I talked to the IT and they state that we only have access with proxy and it seems that adt is not considering a proxy. Is there any way to set proxy properties for adt/java?

**Reply by ajwfrost** [reply]
I thought normally it should pick up the same normal Internet Settings as your browser does...
https://www.java.com/en/download/help/proxy_setup.html

I just hit the Windows button and then typed "Java" into my Windows 10 machine, it came up with the suggestion of "Java Settings" and in the Advanced bit there I found a network settings thing similar to that shown in the above link. If it's possible to set it the same as the browser it's probably more sustainable...

Or you can check your browser settings with start/search for "internet options" from the control panel (or IE -> Settings, if you still have that installed..)

thanks

**Reply by 2jfw** [reply]
Not sure how the mechanics behind this are working but here some further rather strange observations:
- IT gave me a temp whitelisting for no proxy and license check immediately worked
- Restarted PC and whitelisting was removed, however the license check did still succeed
- I deleted the `adt.lic` file from home (users/.airsdk)
- License check failed/timed out with error from above
- Copied valid license file back to home (users/.airsdk)
- License check still failing
- Observation: After this fail, license file was replaced with a new file with different header (#Mon Sep 05 14:32:36 CEST 2022) and developer id and key
- check for the new `adt.lic` file stated "DeveloperID=unknown", before it showed our actual developer id, so I would think a default `adt.lic` file was used to replace our existing one after check fail

Java Proxy settings, as you mentioned, do not seem to affect the the result of the check so I'm looking for a different solution to this. Any other clues or suggestions what I could try out to get the adt check make use of the proxy?

**Reply by ajwfrost** [reply]
Probably some of the behaviour above is because the `adt -license` call won't result in a call to our servers if the license file is already present and valid. It checks only every couple of weeks (unless of course the file is wrong/invalid/out-dated..). One way to force it to check online is to provide the `-licenseDevID` command-line argument, e.g.
```
adt -licenseDevID a-b-c-d -license
```
and this will always try to check online.

One option for setting up a proxy seems to be to add this into the Java command line (so e.g. on Windows, edit the "adt.bat" file):
```
java -Dhttp.proxyHost=127.0.0.1 -Dhttp.proxyPort=8080 ......
```
I also wonder whether it takes into account of the HTTP_PROXY environment variable (`set HTTP_PROXY=127.0.0.1:8080` and `set HTTPS_PROXY=127.0.0.1:8080`, etc..

thanks

**Reply by 2jfw** [reply]
Passing the proxy and port to the `adt.jar` from `adt.bat` did do the trick! - only I had to use `https` instead of `http`.

`adt.bat` then looks like:
`@java -Dhttps.proxyHost=proxy.mycompany.com -Dhttps.proxyPort=8080 -jar "%~dp0\..\lib\adt.jar" %*`

Regarding the System Environmental Variable, the workstations do already have a predefined value from our company and it was already present at the time the tests did fail so I would assume that `HTTPS_PROXY` does **not** work.

Thanks a lot for your support, Andrew! 🍻

**Reply by ajwfrost** [reply]
Great, thanks for helping with all this and good to know we have a way to do it, even if it's a bit of pain to edit the batch file..

If you already had an environment variable for `HTTPS_PROXY` then actually the obvious thing would be for us to read that and manually set up the proxy ... although also ... if you're able to just check, would the below command-line also work for you?

```
@java -Djava.net.useSystemProxies=true -jar "%~dp0\..\lib\adt.jar" %*
```

If so, that seems like a more generic change that we should make globally...

**Reply by 2jfw** [reply]
You're welcome!
So I did check with `-Djava.net.useSystemProxies=true` very carefully on this because first it looked like it worked but I saw the `adt.lic` file was modified after the latest successful check so I replaced it with the "genuine" one and then the check failed.

Let me know if you would like me to test anything further on this..

**Reply by ajwfrost** [reply]
Okay interesting/shame... but that's fine I think, we can look at adding some code to pick up the environment variable manually, and then as a last resort people can edit the batch file..

thanks

**Reply by 2jfw** [reply]
Sounds good, thanks!
Just for completness: I tested `-Djava.net.useSystemProxies=true` again, this time not using the proxy `Script address` but the `Manual proxy setup` with the same result: failed.

**Reply by 2jfw** [reply]
Would passing these arguments to ADL.exe work the same way?
Experiencing the splash in debug mode on IntelliJ and cannot seem to overcome this issue since when adding under **Options / AIR Debug Launcher options** in IntelliJ, ADL throws the following error (due to incorrect occurrence of args):

D:\SDKs\default\bin\adl.exe -profile extendedDesktop -Dhttps.proxyHost=proxy.mycompany.com -Dhttps.proxyPort=8080 D:\gitlab\xxx\xxx\xxx\bin-debug\Main-app.xml D:\gitlab\xxx\xxx\xxx\bin-debug [AIR Debug Launcher]: application descriptor not found [AIR Debug Launcher]: Process finished with exit code 6

**Reply by ajwfrost** [reply]
Hi - no, sorry, ADL is a binary executable so doesn't have the same Java-based arguments. ADL just picks up and inspects the existing license file, it doesn't reach out to online services if it's expired, so probably what you're seeing here is that it's finding an out-dated license file in the "old" location (we moved where ADT looks for/saves these files..).

So if you find this sort of issue, the solution generally is:
a) check the datestamp in the header of the file: ~/.airsdk/adt.lic
b) check the datestamp in the header of the file: AIRSDK_Location/lib/adt.lic
c) if the one in the AIR SDK location is older, then delete that file (or copy the one from the user folder `~/.airsdk` into this lib folder)

thanks

**Reply by 2jfw** [reply]
I actually did check that but both files (`~/.airsdk/adt.lic` and `AIRSDK_Location/lib/adt.lic`) are still the original file which was provided from Harman back in Nov 2021.

Header reads ` #Tue Nov 30 10:06:35 UTC 2021`

I ran the `adt.bat -license` command and it says the file is valid - no changes to the file made on that check so I wonder why the splash does show up. (Also still having the lic file in /bin folder but I guess I can remove that safely now..)

**Reply by 2jfw** [reply]
Got it working by deleting the file in `AIRSDK_Location/lib/adt.lic`  [and also `AIRSDK_Location/bin/adt.lic` (bin folder was likely ignored anyways...)] - Thanks :)

**Reply by ajwfrost** [reply]
Okay strange .. the 2021 file should be immediately overwritten if you use `adt` for anything that requires a license (including the `-license` check)... but yes, deleting that `lib` one is good to do (and for sure, the file under `bin` is not needed).

thanks

**Reply by 2jfw** [reply]
So the license file `~/.airsdk/adt.lic` was overwritten today and now contains a `DeveloperID` and `Key` which is not related to our company and the splash screen shows up again - any clues on why that happened?

**Reply by ajwfrost** [reply]
Overwritten with a different developer ID is concerning (and shouldn't be possible..!) - can you send me the details via direct email? if I can check your developer ID and the new one, I'll see if there's anything that may have happened. But from memory, the ADT tool loads in the file, sends off for a new Key value, but then just re-writes the original DevID back again... 

thanks

**Reply by 2jfw** [reply]
Thanks Andrew, I will send you an email now!

**EDIT:**
I send you a mail with the details.
Having a closer look, the developer ID was actually not overwritten but the Key.
The order/line of `DeveloperID` and `Key` did swap with the overwrite so I misread the values and it looked like it was changed. Sorry for the confusion..

However, the splash now shows up again..

**Reply by 2jfw** [reply]
So, I had changed my AIR SDK back to 795 and in that version there still was the `adt.lic` file in the `lib` folder 😅.
However, after deleting that file and replacing the license file from `~/.airsdk/` with the genuine license file and recompiling the app the splash still shows up - maybe an issue with 795 not considering license file in `~/.airsdk/`?

**Reply by ajwfrost** [reply]
Thanks -> yes exactly, build 795 had a bug there (i.e. our oversight in not handling the new location in ADL...)

From 33.1.1.821:
> github-1238: Ensuring ADL checks under ~/.airsdk for a license file

thanks

**Comment by myflashlab** [comment]
Same problem here. The splash screen started to show up on our desktop software. .apk and .ipa builds seem normal however and the problem seems to be on desktop builds only. I contacted support and they sent me a new license file but still the same problem. I tried SDKs 795, 821 and 856 and no luck yet.

This is becoming pretty urgent as it has took a week and we are stuck by not being able to make a new release to our software

**Reply by ajwfrost** [reply]
When you package up the desktop software using ADT etc, do you see the problem? or is it just when launching the debugger version?

The issue here (normally) is just in ADL, the solution can depend on the version of SDK you have but generally if you copy the `adt.lic` file from your `~/.airsdk/` folder (or `c:\users\username\.airsdk\` on Windows) into the "lib" folder of your AIR SDK installation, it will start working properly again.

thanks

**Reply by myflashlab** [reply]
Right, I tried packaging and the splash screen is not there. However, on Linux, the debugger does not start and we can't run it on linux which is needed of course to make sure the software is working correctly before packaging the linux version.

I have tried various SDK versions (a few minutes ago tried 33.1.1.929) and placed the adt.lic file inside the lib folder next to adt.jar and tried again but still no joy. I can't figure out what I'm doing wrong.

**Reply by myflashlab** [reply]
**Update**: Oh, the .lic in `~/.airsdk/` seems to be updated automatically and when I removed the .lic in the `lib` folder, the splash screen didn't show up again. Great :) I have not tested on Windows or Linux yet. will let you know shortly.

**Reply by myflashlab** [reply]
Ok, I confirm that removing the .lic file from the `lib` folder and having it in `~/.airsdk/` has solved the problem for me on Mac, Windows  and Linux

**Comment by manishpatel0404** [comment]
The above does not seem to solve my problem. I fired up ADL on Windows after a long time and got the splash screen. The ad.lic date shown in both the airsdk and my AIR SDK installation folder shows 8th Feb, 2023. I even tried removing the licence file from my AIR SDK installation folder.

We have a valid licence. Why is the adt.lic date not updated even when my PC is online?

Sorry! Forgot to add that I am using AIR_SDK_33.1.1.935

**Reply by ajwfrost** [reply]
Can you run "adt -license"? ADL doesn't do anything to update an out-of-date license, the assumption (incorrectly it appears!) is that ADT will be run relatively frequently - certainly it's this which is used when you deploy apps, so if you create a Windows bundle, you should find it doesn't have the splash screen and ADL will also then skip it..

thanks

**Reply by manishpatel0404** [reply]
Yep. Creating a Windows bundle for an update I was working on, solved the problem. Now, the adt.lic file in the ~airsdk folder has been updated to the current date. I have also deleted the licence file from my AIR SDK installation folder.

No more splash screen even in ADL. Thanks @ajwfrost
